go/mysql/constants.go

```diff
@@ -605,3 +605,20 @@ func IsConnErr(err error) bool {
 	}
 	return false
 }
+
+// IsSchemaApplyError returns true when given error is a MySQL error applying schema change
+func IsSchemaApplyError(err error) bool {
```

Comment:
Is this error specific to the current use case? I'm wondering why you chose not to use `withddl`.

Reply:
`withddl` is flawed in that it only applies the schema when there's an error with the query. So, in my case, I ran a `SELECT *` on the tablet -- that didn't error, but the caller would error because it can't find expected columns.

What If I wanted to add an index? There's no error in the schema, `withDDL` will not invoke the `ADD INDEX` statement, but my performance degrades.

And on the flip side, what if I actually _dropped_ a column? If my code no longer uses that column, `withDDL` will not see any problem and keep that column around.

Regardless, this new `IsSchemaApplyError` function covers more schema cases than `withDDL` currently does.

Reply:
:+1: